youtube_dl/extractor/mdr.py

```diff
@@ -128,12 +140,15 @@ def _real_extract(self, url):
 
                     f = {
                         'url': video_url,
-                        'format_id': '%s-%d' % (media_type, vbr or abr),
                         'filesize': filesize,
                         'abr': abr,
                         'preference': 1,
                     }
 
+                    if vbr or abr:
+                        f.update({
+                            'format_id': '%s-%d' % (media_type, vbr or abr)})
```

Comment:
This will skip `format_id` completely even if `media_type` is available.

Reply:
What do you suggest we use for `format_id` if neither `vbr` nor `abr` are available?

Reply:
The starting part that is already used for it.

Reply:
I've updated the pull request, and it kinda works. But due to the fact that the `avCustom.xml` behind kika.de's video pages is missing many of the video information parameters, youtube-dl is detecting these videos as audio-only:
```bash
$ python -m youtube_dl -F https://www.kika.de/petronella-apfelmus/sendungen/sendung124504.html
[MDR] 124504: Downloading webpage
[MDR] 124504: Downloading XML
[MDR] 124504: Downloading m3u8 information
[info] Available formats for 124504:
format code  extension  resolution note
HLS-128      mp4        audio only mp4a.40.2@128k
HLS-391      mp4        audio only mp4a.40.2@391k
HLS-596      mp4        audio only mp4a.40.2@596k
HLS-1161     mp4        audio only mp4a.40.2@1161k
HLS-2444     mp4        audio only mp4a.40.2@2444k
MP4-0        mp4        audio only 0.00B
MP4-1        mp4        audio only 0.00B
MP4-2        mp4        audio only 0.00B (best)
```

Reply:
This should be fixed as well.

Reply:
Not really sure how to fix that. This is one XML asset the site currently provides for one video stream:
```xml
<asset>
<profileName>quality = high</profileName>
<fileName/>
<mediaType/>
<frameWidth/>
<frameHeight/>
<bitrateVideo/>
<bitrateAudio/>
<fileSize>0</fileSize>
<flashMediaServerApplicationURL/>
<flashMediaServerURL/>
<adaptiveHttpStreamingRedirectorUrl>
https://zdfvodnone-vh.akamaihd.net/i/meta-files/zdf/smil/m3u8/300/20/10/201019_mo_neu_lot/2/201019_mo_neu_lot.smil/master.m3u8
</adaptiveHttpStreamingRedirectorUrl>
<dynamicHttpStreamingRedirectorUrl/>
<rtspStreamingUrl>
https://nrodlzdf-a.akamaihd.net/none/zdf/20/10/201019_mo_neu_lot/2/201019_mo_neu_lot_808k_p11v15.mp4
</rtspStreamingUrl>
<progressiveDownloadUrl>
https://nrodlzdf-a.akamaihd.net/none/zdf/20/10/201019_mo_neu_lot/2/201019_mo_neu_lot_808k_p11v15.mp4
</progressiveDownloadUrl>
</asset>
```
No information about resolution, bitrate, filesize. Not really much to go on, is there?